File: day1/day1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func find2020(numberList []int) int {
	for _, x := range numberList {
		for _, y := range numberList {
			for _, z := range numberList {
				if x+y+z == 2020 {
					return x * y * z
				}
			}
		}
	}
	return 0
}

func preprocesInput(filename string) []int {
	numberList := make([]int, 0)
	file, _ := os.Open(filename)

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		num, _ := strconv.Atoi(scanner.Text())
		numberList = append(numberList, num)
	}

	return numberList
}

func main() {
	var numberList = preprocesInput("day1_input.txt")
	var answer = find2020(numberList)
	fmt.Println(answer)
}

File: day1/day1.py
example_input = [int(x) for x in "1721 979 366 299 675 1456".split(" ")]

def find_2020(number_list):
    for x in number_list:
        for y in number_list:
            for z in number_list:
                if x + y + z == 2020:
                    return x*y*z

def process_input(filename):
    number_list = list()
    with open(filename) as input_file:
        for line in input_file:
            number_list.append(int(line))
    return number_list

number_list = process_input('day1.in')
print(find_2020(number_list))